backend/src/main/scala/cromwell/backend/async/KnownJobFailureException.scala

```diff
@@ -26,16 +26,22 @@ final case class StderrNonEmpty(jobTag: String, stderrLength: Long, stderrPath:
   override def getMessage = s"stderr for job $jobTag has length $stderrLength and 'failOnStderr' runtime attribute was true."
 }
 
-final case class RetryWithMoreMemory(jobTag: String, stderrPath: Option[Path]) extends KnownJobFailureException {
-  override def getMessage = s"stderr for job $jobTag contained one of the `memory-retry` error-keys specified in the config. " +
-    "Job might have run out of memory."
-}
-
-final case class MemoryMultiplierLessThanOne(jobTag: String,
-                                             stderrPath: Option[Path],
-                                             currentMultiplier: GreaterEqualRefined,
-                                             memoryRetryFactor: GreaterEqualRefined) extends KnownJobFailureException {
-  override def getMessage = s"The result of multiplying current memory multiplier $currentMultiplier with memory retry factor $memoryRetryFactor was not positive."
+final case class RetryWithMoreMemory(jobTag: String,
+                                     stderrPath: Option[Path],
+                                     memoryRetryErrorKeys: Option[List[String]],
+                                     logger: LoggingAdapter) extends KnownJobFailureException {
+  val errorKeysAsString = memoryRetryErrorKeys match {
+    case None =>
+      // this should not occur at this point as one would reach this error class only if Cromwell found one of the
+      // `memory-retry-error-keys` in `stderr` of the task, which is only checked if the `memory-retry-error-keys`
+      // are instantiated in Cromwell config
+      logger.error(s"Programmer error: found one of the `system.memory-retry-error-keys` in the `stderr` of task but " +
+        s"didn't find the error keys while generating the exception!")
+      ""
+    case Some(keys) => s": [${keys.mkString(",")}]"
```

Comment:
I'm pretty sure there's an alternative 3-parameter version of `mkString` which allows you to put the prefix and suffix onto the string as part of the function call (ie rather than requiring this string interpolation)

Reply:
TIL. Yes there is a 3 parameter version of it! Its `mkString(start: String, sep: String, end: String)`